<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Calculate the distance between peak summits and motifs — summit_to_motif • MotifStats</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Calculate the distance between peak summits and motifs — summit_to_motif"><meta property="og:description" content="summit_to_motif() calculates the distance between each motif and its
nearest peak summit. runFimo from the memes package is used to
recover the locations of each motif."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MotifStats</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MotifStats.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/MotifStats/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculate the distance between peak summits and motifs</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/MotifStats/blob/HEAD/R/summit_to_motif.R" class="external-link"><code>R/summit_to_motif.R</code></a></small>
    <div class="hidden name"><code>summit_to_motif.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>summit_to_motif()</code> calculates the distance between each motif and its
nearest peak summit. <code>runFimo</code> from the <code>memes</code> package is used to
recover the locations of each motif.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">summit_to_motif</span><span class="op">(</span></span>
<span>  <span class="va">peak_input</span>,</span>
<span>  <span class="va">motif</span>,</span>
<span>  fp_rate <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  <span class="va">genome_build</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>peak_input</dt>
<dd><p>Either a path to the narrowPeak file or a GRanges peak
object generated by <code><a href="read_peak_file.html">read_peak_file()</a></code>.</p></dd>


<dt>motif</dt>
<dd><p>An object of class <code>PWMatrix</code>.</p></dd>


<dt>fp_rate</dt>
<dd><p>The desired false-positive rate. A p-value threshold will be
selected based on this value. The default false-positive rate is
0.05.</p></dd>


<dt>genome_build</dt>
<dd><p>The genome build that the peak sequences should be
derived from.</p></dd>


<dt>out_dir</dt>
<dd><p>Location to save the 0-order background file. By default, the
background file will be written to a temporary directory.</p></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/memes/man/runFimo.html" class="external-link">memes::runFimo</a></code></p><dl><dt><code>parse_genomic_coord</code></dt>
<dd><p><code>logical(1)</code> whether to parse genomic position
from fasta headers. Fasta headers must be UCSC format positions (ie
"chr:start-end"), but base 1 indexed (GRanges format). If names of fasta
entries are genomic coordinates and parse_genomic_coord == TRUE, results
will contain genomic coordinates of motif matches, otherwise FIMO will return
relative coordinates (i.e. positions from 1 to length of the fasta entry).</p></dd>

    <dt><code>skip_matched_sequence</code></dt>
<dd><p><code>logical(1)</code> whether or not to include the DNA
sequence of the match. Default: <code>FALSE</code>. Note: jobs will complete faster if
set to <code>TRUE</code>. <code>add_sequence()</code> can be used to lookup the sequence after data import if
<code>parse_genomic_coord</code> is <code>TRUE</code>, so setting this flag is not strictly needed.</p></dd>

    <dt><code>max_strand</code></dt>
<dd><p>if match is found on both strands, only report strand with
best match (default: TRUE).</p></dd>

    <dt><code>text</code></dt>
<dd><p><code>logical(1)</code> (default: <code>TRUE</code>). No output files will be created
on the filesystem. The results are unsorted and no q-values are computed.
This setting allows fast searches on very large inputs. When set to <code>FALSE</code>
FIMO will discard 50% of the lower significance matches if &gt;100,000 matches are
detected. <code>text = FALSE</code> will also incur a performance penalty because it
must first read a file to disk, then read it into memory. For these reasons,
I suggest keeping <code>text = TRUE</code>.</p></dd>

    <dt><code>meme_path</code></dt>
<dd><p>path to <code>meme/bin/</code> (optional). Defaut: <code>NULL</code>, searches
"MEME_PATH" environment variable or "meme_path" option for path to "meme/bin/".</p></dd>

    <dt><code>silent</code></dt>
<dd><p><code>logical(1)</code> whether to suppress stdout/stderr printing to
console (default: TRUE). If the command is failing or giving unexpected
output, setting <code>silent = FALSE</code> can aid troubleshooting.</p></dd>

  
</dl></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A list containing an expanded GRanges peak object with metadata
columns relating to motif positions along with a vector of summit-to-motif
distances for each valid peak.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function is designed to work with narrowPeak files from MACS2/3.</p>
<p>To calculate the p-value threshold for a desired false-positive rate, we use
the approximate formula:
$$p \approx \frac{fp\_rate}{2 \times \text{average peak width}}$$</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="https://rdrr.io/pkg/memes/man/runAme.html" class="external-link">runAme</a></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"creb_peaks"</span>, package <span class="op">=</span> <span class="st">"MotifStats"</span><span class="op">)</span> <span class="co"># GRanges</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"creb_motif"</span>, package <span class="op">=</span> <span class="st">"MotifStats"</span><span class="op">)</span> <span class="co"># universalmotif</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">summit_to_motif</span><span class="op">(</span></span></span>
<span class="r-in"><span>  peak_input <span class="op">=</span> <span class="va">creb_peaks</span>,</span></span>
<span class="r-in"><span>  motif <span class="op">=</span> <span class="va">creb_motif</span>,</span></span>
<span class="r-in"><span>  fp_rate <span class="op">=</span> <span class="fl">5e-02</span>,</span></span>
<span class="r-in"><span>  genome_build <span class="op">=</span> <span class="fu">BSgenome.Hsapiens.UCSC.hg38</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.UCSC.hg38/man/package.html" class="external-link">BSgenome.Hsapiens.UCSC.hg38</a></span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Thomas Roberts.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

